src/Tribe/Repositories/Post_Repository.php

```diff
@@ -0,0 +1,245 @@
+<?php
+/**
+ * Handles all of the post querying.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Tickets\Repositories
+ */
+
+namespace Tribe\Tickets\Repositories;
+
+use Tribe\Tickets\Repositories\Traits\Event;
+use Tribe\Tickets\Repositories\Traits\Post_Attendees;
+use Tribe\Tickets\Repositories\Traits\Post_Tickets;
+use Tribe__Repository;
+use Tribe__Timezones as Timezones;
+use Tribe__Utils__Array as Arr;
+
+/**
+ * Class Post_Repository.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Tickets\Repositories
+ */
+class Post_Repository extends Tribe__Repository {
+
+	use Post_Attendees;
+	use Post_Tickets;
+	use Event;
+
+	/**
+	 * The unique fragment that will be used to identify this repository filters.
+	 *
+	 * @var string
+	 */
+	protected $filter_name = 'tickets_post';
+
+	/**
+	 * Post_Repository constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->default_args = array_merge( $this->default_args, [
+			'orderby'                      => 'date',
+			'post_status'                  => 'any',
+			'post_type'                    => 'any',
+			// We'll be handling the dates, let's mark the query as a non-filtered one.
+			'tribe_suppress_query_filters' => true,
+		] );
+
+		$this->schema = array_merge( $this->schema, [
+			// These filter methods are added by the Event trait.
```

Comment:
* :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: 835816ca19fe15a47c7fa21f0e11181e</sub>